sechub-sarif/src/main/java/com/daimler/sechub/sarif/model/Result.java

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.sarif.model;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+/**
+ * /** Reporting descriptor reference, see <a href=
+ * "https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317638">SARIF
+ * 2.1.0 specification entry</a>
+ * 
+ * @author Albert Tregnaghi
+ *
+ */
+@JsonPropertyOrder({ "ruleId", "level", "message", "locations", "properties" })
+public class Result {
+    private String ruleId;
+    private Message message;
+    private Level level;
+
+    private int ruleIndex;
+
+    private List<Location> locations;
+    private List<CodeFlow> codeflows;
+
+    private PropertyBag properties = new PropertyBag();
+
+    public Result() {
+        this(null, null);
+    }
+
+    public Result(String ruleId, Message message) {
+        this.ruleId = ruleId;
+        this.message = message;
+
+        this.locations = new LinkedList<Location>();
+        this.codeflows = new LinkedList<CodeFlow>();
+    }
+
+    public void setRuleIndex(int ruleIndex) {
+        this.ruleIndex = ruleIndex;
+    }
+
+    /**
+     * https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317644
+     * 
+     * @return rule index
+     */
+    public int getRuleIndex() {
+        return ruleIndex;
+    }
+
+    public List<CodeFlow> getCodeFlows() {
+        return codeflows;
+    }
+
+    public void addLocation(Location location) {
+        if (location != null) {
+            this.locations.add(location);
+        }
+    }
+
+    public void addAdditionalProperty(String name, String value) {
```

Comment:
I am having problems to understand: Do you want something like on a set or map implementation where you get a boolean TRUE or the former value back when there was already an existing entry (key) for it? 
The add itself will always be "possible" /"successful" -  so I was a little bit confused.

Reply:
Yes, I mean something like the map implementation. You get a `true` back if the add worked otherwise you get a `false` back. For example, because the key already exists in the map. This avoids exceptions.

```java
        if (name != null && value != null) {
            this.properties.put(name, value);
        }
```

You check the values `name` and `value`. In case, one of them is `null` you don't add the property into the map. As a result, I would expect a `false` from the `addAdditionalProperty`, as the check failed and the property was never added. In case the property was added successfully I expect a `true` back.

Reply:
As we discussed : we remove the complete method, because we have getProperties() which returns the PropertyBag

Reply:
done